File: answer/0206/206.linningmii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """

        if head is None:
            return []

        node = head
        reverse_list = [head.val]

        while node.next:
            node = node.next
            reverse_list.insert(0, node.val)

        return reverse_list

File: answer/0206/206.linningmii.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
function reverseList(head) {
    if (!head || !head.next) {
        return head;
    }

    let cur = head;
    let prev = null;

    while (cur) {
        const next = cur.next
        cur.next = prev;
        prev = cur;
        cur = next;
    }

    return prev
}
